test/helpers/object.test.ts

```diff
@@ -22,6 +22,37 @@ describe("helpers/object", () => {
         });
     });
 
+    describe("#forEachPath()", () => {
+      const obj = {
+          foo: true,
+          bar: {
+              a: true,
+              b: [{
+                  c: true
+              }, true]
+          }
+      };
+      let idx = 0;
+      const targets = [[["foo"], obj, "foo"],
+                     [["bar"], obj, "bar"],
+                     [["bar", "a"], obj.bar, "a"],
+                     [["bar", "b"], obj.bar, "b"],
+                     [["bar", "b", "0"], obj.bar.b, "0"],
+                     [["bar", "b", "0", "c"], obj.bar.b[0], "c"],
+                     [["bar", "b", "1"], obj.bar.b, "1"]];
+      it("should call the callback with the correct parameters", () => {
+        HelperObject.forEachPath(obj, (path, obj, key) => {
```

Comment:
Building on my comment above; we probably don't want to hide the variable `obj` with this parameter.